libs-scala/grpc-reverse-proxy/src/test/scala/com/daml/grpc/ReverseProxySpec.scala

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.grpc
+
+import java.util.concurrent.atomic.AtomicReference
+
+import com.daml.grpc.test.GrpcServer
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall
+import io.grpc.ForwardingServerCallListener.SimpleForwardingServerCallListener
+import io.grpc.health.v1.HealthCheckResponse.ServingStatus
+import io.grpc.health.v1.{HealthCheckRequest, HealthGrpc}
+import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc.{
+  Channel,
+  Metadata,
+  ServerCall,
+  ServerCallHandler,
+  ServerInterceptor,
+  StatusRuntimeException,
+}
+import org.scalatest.Inside
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+final class ReverseProxySpec extends AsyncFlatSpec with Matchers with GrpcServer with Inside {
+
+  import ReverseProxySpec._
+  import Services._
+
+  behavior of "ReverseProxy.create"
+
+  it should "fail if the backend does not support reflection" in withServices(health) { backend =>
+    val proxyBuilder = InProcessServerBuilder.forName(InProcessServerBuilder.generateName())
+    ReverseProxy
+      .create(backend, proxyBuilder)
+      .failed
+      .map(_ shouldBe a[StatusRuntimeException])
+  }
+
+  it should "expose the backend's own services" in withServices(health, reflection) { backend =>
+    val proxyName = InProcessServerBuilder.generateName()
+    val proxyBuilder = InProcessServerBuilder.forName(proxyName)
+    ReverseProxy
+      .create(backend, proxyBuilder)
+      .map { proxyServer =>
+        proxyServer.start()
+        val proxy = InProcessChannelBuilder.forName(proxyName).build()
+        getHealthStatus(backend) shouldEqual getHealthStatus(proxy)
```

Comment:
Maybe also test reflection itself? You claim to expose the backend’s own services but currently you only test that the health endpoint is proxied through not that reflection is proxied through.

Reply:
Makes sense. I was just afraid of using the reflection client to not have weird circular dependencies between components, but probably I can find a way to make some form of simplified call just to make sure that the reflection endpoint is exposed.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8481/commits/a4a7b9021935fa081adcc2f535ca7185dfb0cc12